# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContext2D.cpp
# Commit: fc8eaef017f9
# Full Hash: fc8eaef017f97e24cc0a12ea7b4eb5eef61c8b51
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 16:30:15
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 2 - support vw, vh, vmin and vmax units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180619
# ==============================================================================

diff -r 8a94c155e35c -r fc8eaef017f9 dom/canvas/CanvasRenderingContext2D.cpp
--- a/dom/canvas/CanvasRenderingContext2D.cpp	Mon Sep 11 23:50:11 2023 +0000
+++ b/dom/canvas/CanvasRenderingContext2D.cpp	Mon Sep 11 23:50:12 2023 +0000
@@ -2820,6 +2820,10 @@
   }
   gfx::Size GetSize() const override { return Size(mSize); }
 
+  CSSSize GetCSSViewportSize() const override {
+    return GetCSSViewportSizeFromContext(mPresContext);
+  }
+
  private:
   GeckoFontMetrics GetFontMetricsForType(Type aType) const override {
     switch (aType) {